src/utils/utils.py

```diff
@@ -299,63 +269,66 @@ def tags_to_ixs(tags, tag_to_ix_dict, ix_to_tag_dic):
 def load_tsv(
         data_file,
         max_seq_len,
+        label_idx=2,
         s1_idx=0,
         s2_idx=1,
-        targ_idx=2,
-        idx_idx=None,
-        targ_map=None,
-        targ_fn=None,
+        label_fn=None,
         skip_rows=0,
+        return_indices=False,
         delimiter='\t',
         filter_idx=None,
+        has_labels=True,
         filter_value=None):
-    '''Load a tsv
-
-    To load only rows that have a certain value for a certain column, like genre in MNLI, set filter_idx and filter_value.'''
-    sent1s, sent2s, targs, idxs = [], [], [], []
-    with codecs.open(data_file, 'r', 'utf-8', errors='ignore') as data_fh:
-        for _ in range(skip_rows):
-            data_fh.readline()
-        for row_idx, row in enumerate(data_fh):
-            try:
-                row = row.strip().split(delimiter)
-                if filter_idx and row[filter_idx] != filter_value:
-                    continue
-                sent1 = process_sentence(row[s1_idx], max_seq_len)
-                if (targ_idx is not None and not row[targ_idx]) or not len(sent1):
-                    continue
-
-                if targ_idx is not None:
-                    if targ_map is not None:
-                        targ = targ_map[row[targ_idx]]
-                    elif targ_fn is not None:
-                        targ = targ_fn(row[targ_idx])
-                    else:
-                        targ = int(row[targ_idx])
-                else:
-                    targ = 0
-
-                if s2_idx is not None:
-                    sent2 = process_sentence(row[s2_idx], max_seq_len)
-                    if not len(sent2):
-                        continue
-                    sent2s.append(sent2)
-
-                if idx_idx is not None:
-                    idx = int(row[idx_idx])
-                    idxs.append(idx)
-
-                sent1s.append(sent1)
-                targs.append(targ)
-
-            except Exception as e:
-                print(e, " file: %s, row: %d" % (data_file, row_idx))
-                continue
-
-    if idx_idx is not None:
-        return sent1s, sent2s, targs, idxs
+    '''
+    Load a tsv.
+    To load only rows that have a certain value for a certain column,
+    like genre in MNLI, set filter_idx and filter_value.
+    Args:
+
+        s1_idx; int
+        s2_idx: int
+        targ_idx: int
+        return_indices: bool that describes if you need to return indices (for purposes of matching)
+        label_fn is a function that expects a row and outputs the label
+
+    Returns:
+        List of first and second sentences, labels, and if applicable indices
+    '''
+    sent1s, sent2s, labels = pd.Series(), pd.Series(), pd.Series()
+    rows = pd.read_csv(data_file, \
+                        sep=delimiter, \
+                        error_bad_lines=False, \
+                        header=None, \
+                        skiprows=skip_rows, \
+                        quoting=csv.QUOTE_NONE,\
```

Comment:
Add a comment about this choice, and do some quick testing to make sure this isn't a problem on any of the existing data files—it's the kind of thing that can easily mess us up.